renku/models/migrations/__init__.py

```diff
@@ -15,8 +15,14 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Renku migrations."""
+"""Renku JSON-LD migrations."""
 
-from renku.models.migrations.dataset import migrate_dataset
+from renku.models.migrations.dataset import (
+    migrate_dataset_schema,
+    migrate_absolute_paths,
+)
 
-MIGRATIONS = [migrate_dataset]
+JSONLD_MIGRATIONS = {
+    'dctypes:Dataset': [migrate_dataset_schema, migrate_absolute_paths],
```

Comment:
this will fail if the expanded form of the URI is used, e.g. this could be just `http://purl.org/dc/dcmitypes/Dataset` if it came out of a sparql query. We should address this in another PR.

Reply:
True, except @type attribute seems to always be in human-readable format due to compaction. Could you provide us with a concrete example where that would be the case so we can add it as a test?

Reply:
If the context doesn't include the prefix, or if it includes the prefix by a different name, e.g. I would be free to define `schemaorg: http://schema.org` then this breaks. The full solution is to use the jsonld library here instead of a simple string match IMO. That's why I suggested we handle it in a follow-up issue.